metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -54,7 +55,9 @@ def get_records(self, workunit: WorkUnitType) -> Iterable[RecordEnvelope]:
 # See https://github.com/python/mypy/issues/5374 for why we suppress this mypy error.
 @dataclass  # type: ignore[misc]
 class Source(Closeable, metaclass=ABCMeta):
-    ctx: PipelineContext
+    def __init__(self, ctx: PipelineContext) -> None:
```

Comment:
This is a dataclass! We can either get rid of the annotation or the __init__ method.